spring-cloud-netflix-eureka-server/src/main/java/org/springframework/cloud/netflix/eureka/server/InstanceRegistry.java

```diff
@@ -78,43 +79,39 @@ public void openForTraffic(ApplicationInfoManager applicationInfoManager, int co
 
 	@Override
 	public void register(InstanceInfo info, int leaseDuration, boolean isReplication) {
-		handleRegistration(info, leaseDuration, isReplication);
 		super.register(info, leaseDuration, isReplication);
+		handleRegistration(info, leaseDuration, isReplication);
 	}
 
 	@Override
 	public void register(final InstanceInfo info, final boolean isReplication) {
-		handleRegistration(info, resolveInstanceLeaseDuration(info), isReplication);
 		super.register(info, isReplication);
+		handleRegistration(info, resolveInstanceLeaseDuration(info), isReplication);
 	}
 
 	@Override
 	public boolean cancel(String appName, String serverId, boolean isReplication) {
+		final boolean cancelled = super.cancel(appName, serverId, isReplication);
 		handleCancelation(appName, serverId, isReplication);
```

Comment:
Since we get a boolean, we might only want to  handle cancellation if `cancelled` is `true`.

Reply:
That seems correct.
I'll change renew in the same way.